priv/riak_kv.schema

```diff
@@ -968,44 +968,48 @@
      (_) -> false
   end}.
 
-%% @doc Register allowable origins (for CORS), specified as a list of ip
-%% ({host} or {host, port} tuple) under a scheme, i.e.
+%% @doc listener.$scheme.cors_allowable_origin.$origin is an IP address and TCP
+%% port that is allowed access to resources via Cross-Origin Resource Sharing
+%% (CORS).
 %%
-%% {riak_kv, [
-%%   {allowable_origin, {
-%%      http, [
-%%          {"127.0.0.1", 3000},
-%%          {"10.10.1.240"}
-%%      ],
-%%      https, [
-%%          {"127.0.0.1", 4430}
-%%      ]
-%%   }}
-%% ]}
-%%
-%% The resulting allowable origins are "http://127.0.0.1:300",
-%% "http://10.10.1.240:*", "https://127.0.0.1:4430".
-%%
-%% The special case of "Allow All" is achieved by specifying the scheme as '*'
-%% (wildcard), i.e.
-%%
-%% {riak_kv, [
-%%   {allowable_origin, { '*' } }
-%% ]}
-%%
-%% If port is not specified, all ports for the host are to be considered allowed.
-%% This is equivalent to specifying the port using the wildcard.
-{mapping, "allowable_origin.$name", "riak_kv.allowable_origin.http", [
-    {datatype, ip},
-    {default, {}},
-    {level, advanced}
+%% Allowable schemes include http and https, and all where the wildcard (all)
+%% is used to allow all origins to share resources with Riak.
+{mapping, "listener.$scheme.cors_allowable_origin.$origin", "riak_api.cors_allowable_origins",
+[
+ {datatype, [ip, flag]}
```

Comment:
What do `on` and `off` mean as values here? 

Tried setting this in `riak.conf`:

```
listener.http.cors_allowable_origin.foo = off
```

Got this failure:

```09:31:11.691 [error] CRASH REPORT Process <0.364.0> with 0 neighbours exited with reason: {{{badmatch,false},[{riak_api_web,binding_config,2,[{file,"src/riak_api_web.erl"},{line,48}]},{riak_api_sup,'-web_processes/1-lc$^0/1-0-',1,[{file,"src/riak_api_sup.erl"},{line,63}]},{riak_api_sup,web_processes,1,[{file,"src/riak_api_sup.erl"},{line,63}]},{riak_api_sup,init,1,[{file,"src/riak_api_sup.erl"},{line,53}]},{supervisor,init,1,[{file,"supervisor.erl"},{line,239}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,304}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,239}]}]},...} in application_master:init/4 line 133
09:31:11.691 [info] Application riak_api exited with reason: {{{badmatch,false},[{riak_api_web,binding_config,2,[{file,"src/riak_api_web.erl"},{line,48}]},{riak_api_sup,'-web_processes/1-lc$^0/1-0-',1,[{file,"src/riak_api_sup.erl"},{line,63}]},{riak_api_sup,web_processes,1,[{file,"src/riak_api_sup.erl"},{line,63}]},{riak_api_sup,init,1,[{file,"src/riak_api_sup.erl"},{line,53}]},{supervisor,init,1,[{file,"supervisor.erl"},{line,239}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,304}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,239}]}]},...}
```